reana_client/validation/complexity.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2021 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA workflow complexity estimation."""
+
+import os
+
+from reana_client.printer import display_message
+
+
+def estimate_complexity(workflow_type, reana_yaml):
+    """Estimate complexity in REANA workflow.
+
+    :param workflow_type: A supported workflow specification type.
+    :param reana_yaml: REANA YAML specification.
+    """
+
+    def build_estimator(workflow_type, reana_yaml):
+        if workflow_type == "serial":
+            return SerialComplexityEstimator(reana_yaml)
+        if workflow_type == "yadage":
+            return YadageComplexityEstimator(reana_yaml)
+        if workflow_type == "cwl":
```

Comment:
You may want to use if/elif/elif/else to take care of possibly new workflow types, e.g. Snakemake is forthcoming...